src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs

```diff
@@ -451,6 +450,8 @@ private async Task ExecuteInitScriptsAsync()
 
                     return;
                 }
+                // We may be enumerating packages from disk here. Always do it from a background thread.
+                await TaskScheduler.Default;
```

Comment:
Did you test a package that uses init.ps1 or install.ps1, that uses the $dte or $project objects?  I'm pretty sure they must run on the UI thread.

Reply:
This only handles init scripts, which are always run from a threadpool thread: https://github.com/NuGet/NuGet.Client/blob/229fe85a271e9c059d14849d6e9fcc833e61809c/src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs#L492-L493